Source/WebCore/rendering/RenderTreeAsText.cpp

```diff
@@ -649,6 +651,11 @@ void write(TextStream& ts, const RenderObject& o, OptionSet<RenderAsTextFlag> be
                 if (RenderLayer* layer = root->layer())
                     writeLayers(ts, *layer, *layer, layer->rect(), behavior);
             }
+        } else if (is<RemoteFrameView>(widget)) {
```

Comment:
Let's make this a pure virtual function on FrameView and have each type of FrameView implement its own functionality.  That way we don't have lots of places where we have the "if (local) else if (remote)" pattern.

Reply:
that's fair, but I'm worried that it might change what we do for LocalFrameView currently because the only way would be for us to call externalRepresentation() LocalFrameView.